nsxt/resource_nsxt_policy_tier0_gateway_ha_vip_config.go

```diff
@@ -0,0 +1,364 @@
+/* Copyright © 2019 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	gm_tier0s "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-gm/global_infra/tier_0s"
+	gm_model "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-gm/model"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/tier_0s"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	"strconv"
+	"strings"
+)
+
+var interfacePathLen = 8
+var interfaceTier0Location = 3
+var interfaceLocaleSrvLocation = 5
+
+func resourceNsxtPolicyTier0GatewayHAVipConfig() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyTier0GatewayHAVipConfigCreate,
+		Read:   resourceNsxtPolicyTier0GatewayHAVipConfigRead,
+		Update: resourceNsxtPolicyTier0GatewayHAVipConfigUpdate,
+		Delete: resourceNsxtPolicyTier0GatewayHAVipConfigDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceNsxtPolicyTier0GatewayHAVipConfigImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"config": {
+				Type:        schema.TypeList,
+				Description: "Tier0 HA VIP Config",
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:        schema.TypeBool,
+							Description: "Flag to enable this HA VIP config",
+							Optional:    true,
+							Default:     true,
+						},
+						"external_interface_paths": {
+							Type:        schema.TypeList,
+							Description: "paths to Tier0 external interfaces which are to be paired to provide redundancy",
+							Required:    true,
+							MinItems:    2,
+							MaxItems:    2,
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validatePolicyPath(),
+							},
+						},
+						"vip_subnets": {
+							Type:        schema.TypeList,
+							Description: "IP address subnets which will be used as floating IP addresses",
+							Required:    true,
+							MinItems:    1,
+							MaxItems:    2,
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validateIPCidr(),
+							},
+						},
+					},
+				},
+			},
+			"locale_service_id": {
+				Type:        schema.TypeString,
+				Description: "Id of associated Gateway Locale Service on NSX",
+				Computed:    true,
+			},
+			"tier0_id": {
```

Comment:
I don't think we need both locale service and gw path, since dependency need should be satisfied by locale service path.